### PR TITLE
Convert the audio quality details popover to shadcn

### PR DESCRIPTION
**Pull request by marcelveldt**
The stream quality details popover was still built with Vuetify (`v-menu`, `v-card`, `v-chip`, `v-tooltip`, mdi icons), so it looked out of place next to the fullscreen player and queue that have already moved to shadcn-vue.

This converts it fully to shadcn-vue (reka-ui) + lucide icons so it matches the surrounding UI, and tidies up the audio signal-chain diagram in the process.

- Replace the Vuetify menu/card/list/chip/tooltip/icons with shadcn `Popover`, `Button` and `Tooltip` plus lucide icons
- Retokenise the signal-flow diagram to the shadcn theme variables so it themes correctly in light and dark
- Compact and align the layout, and size the popover to its content so it no longer wraps on wide displays
- Separate multiple output blocks with a dotted connector so each reads as its own branch off the shared source
- Hide the codec row when the codec is unknown and show the full codec info in the output-format tooltip

### CODE REVIEW

File: src/components/QualityDetailsBtn.vue [hunk 1]
```diff
@@ -435,64 +363,55 @@
...
               >
-                <img
-                  class="streamdetails-icon invert-on-dark-mode"
-                  src="@/assets/dsp.svg"
-                  alt=""
-                />
+                <SlidersHorizontal :size="22" class="streamdetails-glyph" />
                 {{
                   $t("streamdetails.output_gain", [dsp.input_gain.toFixed(1)])
                 }}
```
[the remark this answers is not in the document]
Comment (marcelveldt) [reply src/components/QualityDetailsBtn.vue]: Good catch - this was a pre-existing bug carried over from the Vuetify version. Fixed to read the output gain instead of the input gain.

File: src/components/QualityDetailsBtn.vue [hunk 2]
```diff
@@ -269,135 +193,143 @@
...
-                    streamDetails.audio_format.channels,
-                  ])
-                }}
-                <span v-if="streamDetails.audio_format.bit_rate">
-                  <br />
+              <Tooltip>
+                <TooltipTrigger as-child>
+                  <button type="button" class="streamdetails-info">
+                    <Info :size="16" />
+                  </button>
```
[the remark this answers is not in the document]
Comment (marcelveldt) [reply src/components/QualityDetailsBtn.vue]: Added an aria-label here (new more_info translation key) so screen readers announce the button.

File: src/components/QualityDetailsBtn.vue [hunk 3]
```diff
@@ -269,135 +193,143 @@
...
+            </div>
+            <!-- Volume Normalization -->
+            <div v-if="loudness" class="streamdetails-item">
+              <AudioLines :size="22" class="streamdetails-glyph" />
+              {{ loudness }}
+              <Tooltip>
+                <TooltipTrigger as-child>
+                  <button type="button" class="streamdetails-info">
+                    <Info :size="16" />
+                  </button>
```
[the remark this answers is not in the document]
Comment (marcelveldt) [reply src/components/QualityDetailsBtn.vue]: Added an aria-label here (new more_info translation key) so screen readers announce the button.

File: src/components/QualityDetailsBtn.vue [hunk 4]
```diff
@@ -435,64 +363,55 @@
...
-                      >mdi-information</v-icon
-                    >
-                  </template>
-                  {{ $t("streamdetails.dsp_disabled_by_unsupported_group") }}
-                </v-tooltip>
+                <Tooltip>
+                  <TooltipTrigger as-child>
+                    <button type="button" class="streamdetails-info">
+                      <Info :size="16" />
+                    </button>
```
[the remark this answers is not in the document]
Comment (marcelveldt) [reply src/components/QualityDetailsBtn.vue]: Added an aria-label here (new more_info translation key) so screen readers announce the button.

File: src/components/QualityDetailsBtn.vue [hunk 5]
```diff
@@ -435,64 +363,55 @@
...
-                      >mdi-information</v-icon
-                    >
-                  </template>
-                  {{ $t("streamdetails.output_limiter_info") }}
-                </v-tooltip>
+                <Tooltip>
+                  <TooltipTrigger as-child>
+                    <button type="button" class="streamdetails-info">
+                      <Info :size="16" />
+                    </button>
```
[the remark this answers is not in the document]
Comment (marcelveldt) [reply src/components/QualityDetailsBtn.vue]: Added an aria-label here (new more_info translation key) so screen readers announce the button.

File: src/components/QualityDetailsBtn.vue [hunk 6]
```diff
@@ -516,31 +435,66 @@
...
-                    >
-                  </template>
-                  <template
-                    v-if="
-                      isPcm(
+                <Tooltip>
+                  <TooltipTrigger as-child>
+                    <button type="button" class="streamdetails-info">
+                      <Info :size="16" />
+                    </button>
```
[the remark this answers is not in the document]
Comment (marcelveldt) [reply src/components/QualityDetailsBtn.vue]: Added an aria-label here (new more_info translation key) so screen readers announce the button.

File: src/components/QualityDetailsBtn.vue [hunk 7]
```diff
@@ -591,30 +546,36 @@
...
-                          >mdi-information</v-icon
-                        >
-                      </template>
-                      {{ $t("streamdetails.click_to_expand") }}
-                    </v-tooltip>
+                    <Tooltip>
+                      <TooltipTrigger as-child>
+                        <button type="button" class="streamdetails-info">
+                          <Info :size="16" />
+                        </button>
```
[the remark this answers is not in the document]
Comment (marcelveldt) [reply src/components/QualityDetailsBtn.vue]: Added an aria-label here (new more_info translation key) so screen readers announce the button.

File: src/components/QualityDetailsBtn.vue [hunk 8]
```diff
@@ -706,6 +673,24 @@ const dsp_grouped = computed(() => {
...
+  if (dsp.output_gain && dsp.output_gain != 0) return "output_gain";
+  if (dsp.output_limiter) return "output_limiter";
+  if (
+    streamDetails.value &&
+    streamDetails.value.dsp &&
+    streamDetails.value.dsp[player_id].output_format
+  )
+    return "output_format";
+  return "player";
+};
```
[the remark this answers is not in the document]
Comment (marcelveldt) [reply src/components/QualityDetailsBtn.vue]: Good call - split the diagram out into its own StreamDetailsChain component and moved the shared quality-tier logic into a useStreamQuality composable, so this file is now just the trigger + popover.

File: src/composables/useStreamQuality.ts [hunk 9]
```diff
@@ -0,0 +1,169 @@
...
+      return tiers;
+    }
+    const inputTier = inputQualityTier.value;
+    for (const player_id of Object.keys(sd.dsp)) {
+      // Output quality can never be higher than input quality
+      tiers[player_id] = Math.min(
+        outputQualityTiers.value[player_id],
+        inputTier,
+      );
+    }
```
[the remark this answers is not in the document]
Comment (marcelveldt) [reply src/composables/useStreamQuality.ts:152]: Fixed - default to UNKNOWN when a player has no known output format, so this can no longer produce NaN.

File: src/composables/useStreamQuality.ts [hunk 10]
```diff
@@ -0,0 +1,169 @@
...
+    return tiers;
+  });
+
+  const minOutputQualityTier = computed(() => {
+    return Math.min(...Object.values(combinedOutputQualityTiers.value));
+  });
```
[the remark this answers is not in the document]
Comment (marcelveldt) [reply src/composables/useStreamQuality.ts:159]: Fixed - returns UNKNOWN when there are no output tiers, instead of Infinity.

File: src/composables/useStreamQuality.ts [hunk 11]
```diff
@@ -0,0 +1,169 @@
...
+
+  const maxOutputQualityTier = computed(() => {
+    return Math.max(...Object.values(combinedOutputQualityTiers.value));
+  });
```
[the remark this answers is not in the document]
Comment (marcelveldt) [reply src/composables/useStreamQuality.ts:164]: Fixed - returns UNKNOWN when there are no output tiers, instead of -Infinity.

File: src/composables/useStreamQuality.ts [hunk 12]
```diff
@@ -0,0 +1,169 @@
...
+      return "cyan";
+    default:
+      return "gray";
+  }
+}
+
+export function useStreamQuality(
+  streamDetails: MaybeRefOrGetter<StreamDetails | undefined>,
+) {
+  const inputQualityTier = computed(() => {
```
[the remark this answers is not in the document]
Comment (marcelveldt) [reply src/composables/useStreamQuality.ts:82]: Added tests/composables/useStreamQuality.test.ts covering the tier logic plus the empty / unknown-output edge cases.

File: src/translations/en.json [hunk 13]
```diff
@@ -767,6 +767,7 @@
     "streamdetails": {
         "input_header": "Input",
         "output_header": "Output",
+        "more_info": "More information",
         "output_format_info": "Audio is sent in the {0} format",
```
[the remark this answers is not in the document]
Comment (marcelveldt) [reply src/translations/en.json:772]: Translations are Lokalise-managed here - we only edit en.json and the other locale files are synced from it (missing keys fall back to en), so I have not added it to the others by hand.

File: src/components/StreamDetailsChain.vue [hunk 14]
```diff
@@ -0,0 +1,951 @@
...
+          >
+            <div
+              v-for="(player, i) in players"
+              :key="i"
+              class="streamdetails-item"
+              @click="is_dsp_group_expanded[player_id] = false"
+            >
+              <template v-if="api.players[player]">
+                <ProviderIcon
+                  :domain="outputProtocolDomain(api.players[player_id])"
```
[the remark this answers is not in the document]
Comment (marcelveldt) [reply src/components/StreamDetailsChain.vue]: Fixed - the expanded rows now derive the icon from the actual player instead of the group key.

File: src/components/StreamDetailsChain.vue [hunk 15]
```diff
@@ -0,0 +1,951 @@
...
+              player_id in is_dsp_group_expanded &&
+              is_dsp_group_expanded[player_id]
+            "
+          >
+            <div
+              v-for="(player, i) in players"
+              :key="i"
+              class="streamdetails-item"
+              @click="is_dsp_group_expanded[player_id] = false"
+            >
```
[the remark this answers is not in the document]
Comment (marcelveldt) [reply src/components/StreamDetailsChain.vue:481]: Fixed - added role=button, tabindex and Enter/Space handlers to these rows.

File: src/components/StreamDetailsChain.vue [hunk 16]
```diff
@@ -0,0 +1,951 @@
...
+                />
+                Player not found
+              </template>
+            </div>
+          </template>
+          <div
+            v-else
+            class="streamdetails-item"
+            @click="is_dsp_group_expanded[player_id] = true"
+          >
```
[the remark this answers is not in the document]
Comment (marcelveldt) [reply src/components/StreamDetailsChain.vue:510]: Fixed - added role=button, tabindex, aria-expanded and keyboard handlers here too.

File: src/components/StreamDetailsChain.vue [hunk 17]
```diff
@@ -0,0 +1,951 @@
...
+                />
+                {{ api.players[player].name }}
+              </template>
+              <template v-else>
+                <!-- This should not happen -->
+                <AlertCircle
+                  :size="22"
+                  class="streamdetails-glyph text-muted-foreground"
+                />
+                Player not found
```
[the remark this answers is not in the document]
Comment (marcelveldt) [reply src/components/StreamDetailsChain.vue]: Fixed - now uses a streamdetails.player_not_found i18n key.

File: src/components/StreamDetailsChain.vue [hunk 18]
```diff
@@ -0,0 +1,951 @@
...
+                </Tooltip>
+              </template>
+            </template>
+            <template v-else>
+              <!-- This should not happen -->
+              <AlertCircle
+                :size="22"
+                class="streamdetails-glyph text-muted-foreground"
+              />
+              Player not found
```
[the remark this answers is not in the document]
Comment (marcelveldt) [reply src/components/StreamDetailsChain.vue]: Fixed - localized via the same streamdetails.player_not_found key.

File: src/components/StreamDetailsChain.vue [hunk 19]
```diff
@@ -0,0 +1,951 @@
...
+        <div class="line-straight"></div>
+        <!-- Volume Normalization -->
+        <div v-if="loudness" class="line-with-dot"></div>
+        <!-- into the first output section (stays connected to the input) -->
+        <div class="line-straight-halve"></div>
+        <!-- Player -->
+        <template
+          v-for="({ dsp, player_id, players }, index) in dsp_grouped"
+          :key="index"
+        >
```
[the remark this answers is not in the document]
Comment (marcelveldt) [reply src/components/StreamDetailsChain.vue:23]: Fixed - keyed by player_id now.

File: src/components/StreamDetailsChain.vue [hunk 20]
```diff
@@ -0,0 +1,951 @@
...
+                <br />
+              </span>
+            </TooltipContent>
+          </Tooltip>
+        </div>
+
+        <template
+          v-for="({ dsp, player_id, players }, index) in dsp_grouped"
+          :key="index"
+        >
```
[the remark this answers is not in the document]
Comment (marcelveldt) [reply src/components/StreamDetailsChain.vue:290]: Fixed - keyed by player_id now.

File: src/components/StreamDetailsChain.vue [hunk 21]
```diff
@@ -0,0 +1,951 @@
+<template>
+  <!-- audio signal-chain diagram: the left column is the connector "rail"
+       (curves/dots/dotted branches), the right column is each stage of the
+       chain (input -> processing -> per-player output). The two columns share
+       the --sd-row rhythm so rows and rail stay aligned. -->
```
[the remark this answers is not in the document]
Comment (marcelveldt) [reply src/components/StreamDetailsChain.vue:5]: Kept as-is for now: the diagram is one cohesive unit and the meaningful split (trigger vs diagram vs the shared tier logic in useStreamQuality) is already done. The global CSS is intentional since PopoverContent is teleported to body. Can decompose further in a follow-up if it grows.
